packages/block-library/src/image/edit.js

```diff
@@ -87,80 +87,73 @@ const isTemporaryImage = ( id, url ) => ! id && isBlobURL( url );
  */
 const isExternalImage = ( id, url ) => url && ! id && ! isBlobURL( url );
 
-export class ImageEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.updateAlt = this.updateAlt.bind( this );
-		this.updateAlignment = this.updateAlignment.bind( this );
-		this.onFocusCaption = this.onFocusCaption.bind( this );
-		this.onImageClick = this.onImageClick.bind( this );
-		this.onSelectImage = this.onSelectImage.bind( this );
-		this.onSelectURL = this.onSelectURL.bind( this );
-		this.updateImage = this.updateImage.bind( this );
-		this.onSetHref = this.onSetHref.bind( this );
-		this.onSetTitle = this.onSetTitle.bind( this );
-		this.getFilename = this.getFilename.bind( this );
-		this.onUploadError = this.onUploadError.bind( this );
-		this.onImageError = this.onImageError.bind( this );
-
-		this.state = {
-			captionFocused: false,
-		};
+function getFilename( url ) {
+	const path = getPath( url );
+	if ( path ) {
+		return last( path.split( '/' ) );
 	}
+}
 
-	componentDidMount() {
-		const { attributes, mediaUpload, noticeOperations } = this.props;
-		const { id, url = '' } = attributes;
+export function ImageEdit( {
+	attributes: {
+		url = '',
+		alt,
+		caption,
+		align,
+		id,
+		href,
+		rel,
+		linkClass,
+		linkDestination,
+		title,
+		width,
+		height,
+		linkTarget,
+		sizeSlug,
+	},
+	setAttributes,
+	isSelected,
+	className,
+	noticeUI,
+	insertBlocksAfter,
+	noticeOperations,
+	onReplace,
+} ) {
+	const selected = useSelect( ( select ) => {
```

Comment:
You could use destructuring right here rather than destructuring `selected` immediately after.

Reply:
This has been discussed numerous times and I'm not sure if we reached a conclusion on what to do in case of clashing variable names. Cc @youknowriad

Reply:
My way of doing it is to rename the variables inside the useSelector callback because that's the "temporary" part and I  see the returned values as a bit more important to get semantically right.

I also like  inline  useSelect callbacks because the separate selector function is not something that work  consistently and you lose some context.

Reply:
I prefix the temporary variables with an underscore, e.g. `_thing`; that's the approach I'm using in #21427.

Reply:
[Problem solved](
4251d86).